uptane/clients/primary.py

```diff
@@ -619,37 +620,41 @@ def primary_update_cycle(self):
         self.updater.download_target(target, full_targets_directory)
 
       except tuf.NoWorkingMirrorError as e:
-        print('')
-        print(YELLOW + 'In downloading target ' + repr(filepath) + ', am unable '
-            'to find a mirror providing a trustworthy file.\nChecking the mirrors'
-            ' resulted in these errors:')
+        error_report = ''
         for mirror in e.mirror_errors:
-          print('    ' + type(e.mirror_errors[mirror]).__name__ + ' from ' + mirror)
-        print(ENDCOLORS)
-
-        print_banner(BANNER_DEFENDED, color=WHITE+DARK_BLUE_BG,
-            text='No image was found that exactly matches the signed metadata '
-            'from the Director and Image Repositories. Not keeping '
-            'untrustworthy files. ' + repr(target_filepath), sound=TADA)
-        import time
-        time.sleep(3)
+          error_report += \
+              type(e.mirror_errors[mirror]).__name__ + ' from ' + mirror + '; '
+        log.info(YELLOW + 'In downloading target ' + repr(filepath) +
+            ', am unable to find a mirror providing a trustworthy file. '
+            'Checking the mirrors resulted in these errors:  ' + error_report +
+            ENDCOLORS)
+
+        # If running the demo, display splash banner indicating the rejection.
+        # This clause should be pulled out of the reference implementation when
+        # possible.
+        if uptane.DEMO_MODE: # pragma: no cover
+          print_banner(BANNER_DEFENDED, color=WHITE+DARK_BLUE_BG,
+              text='No image was found that exactly matches the signed metadata '
+              'from the Director and Image Repositories. Not keeping '
+              'untrustworthy files. ' + repr(target_filepath), sound=TADA)
+          import time
```

Comment:
From our code style guidelines:

> Imports are always put at the top of the file, just after any module comments and docstrings, and before module globals and constants.

Reply:
I saw `import time` multiple times on separate lines.

Reply:
Yes - see comment in the other PR, [here](https://github.com/uptane/uptane/pull/133#discussion_r139813407).

Reply:
Gonna fix both anyway - I'll have the demo-conditional import at the top and make the comments there.